File: apps/pc-keiba-viewer/src/scripts/running_style_feature_version.py
"""Running-style feature version SSoT loader (reads JSON shared with TS)."""

from __future__ import annotations

import json
from pathlib import Path

_JSON_PATH: Path = (
    Path(__file__).parent / "finish-position-features" / "running-style-feature-version.json"
)
_DATA: dict[str, str] = json.loads(_JSON_PATH.read_text(encoding="utf-8"))

RUNNING_STYLE_FEATURE_VERSION: str = _DATA["version"]
RUNNING_STYLE_FEATURE_VERSION_DESC: str = _DATA["description"]

File: apps/pc-keiba-viewer/src/scripts/finish-position-features/running-style-feature-version.ts
// Run with: imported via bun runtime (vitest / generate-running-style-local.ts)
import runningStyleFeatureVersionJson from "./running-style-feature-version.json" with { type: "json" };

interface RunningStyleFeatureVersionJson {
  version: string;
  description: string;
}

const data: RunningStyleFeatureVersionJson = runningStyleFeatureVersionJson;

export const RUNNING_STYLE_FEATURE_VERSION: string = data.version;
export const RUNNING_STYLE_FEATURE_VERSION_DESC: string = data.description;
